ansible_base/authentication/authenticator_plugins/oidc.py

```diff
@@ -215,3 +217,37 @@ def extra_data(self, user, backend, response, *args, **kwargs):
 
     def get_user_groups(self, extra_groups=[]):
         return extra_groups
+
+    def public_key(self):
+        key = self.setting("PUBLIC_KEY")
+        return (
+            "\n".join(
+                [
+                    "-----BEGIN PUBLIC KEY-----",
+                    key,
+                    "-----END PUBLIC KEY-----",
+                ]
+            )
+            if key
+            else None
+        )
+
+    def user_data(self, access_token, *args, **kwargs):
```

Comment:
Can we add a comment summarizing what we are doing here? (feel free to correct this if needed)
```suggestion
    def user_data(self, access_token, *args, **kwargs):
       """
       This overrides the the user_data function provided by the social auth class OpenIdConnectAuth
       The function in that class assumes the request cming back from userinfo_url is not an encrypted token.
       We want to assume that if we get back a Content-Type of application/jwt that we have an encrypted token.
       So we will decrypt the token and return that data or we will raise a PyJWTError that we can't decrypt the token
       If we don't get that Content-Type back, we will just return whatever data was given to us just like the parent method currently does. 
       """
```

Reply:
Yeah sure, thanks.